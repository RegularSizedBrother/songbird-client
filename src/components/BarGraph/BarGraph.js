import React from 'react';
import {
  BarChart, Bar, XAxis, YAxis, CartesianGrid, Tooltip, ResponsiveContainer, ReferenceLine
} from 'recharts';
import './styles.css';

export default function BarGraph(props) {
  return(
    <ResponsiveContainer width='100%' height={500}>
      <BarChart
        data={props.data}
        margin={{
          top: 40, right: 30, left: 20, bottom: 5,
        }}
      >
        <CartesianGrid strokeDasharray="3 3" />
        <XAxis xAxisId='opposite' dataKey="opposite" orientation='bottom' />
        <XAxis xAxisId='label' dataKey="label" orientation='top' />
        <YAxis hide={true} type='number' domain={[-50, 50]} />
        <ReferenceLine xAxisId='label' y={0} stroke='#000' />
        <Bar xAxisId='label' dataKey="value" fill="#8884d8" />
      </BarChart>
    </ResponsiveContainer>
  );
}
