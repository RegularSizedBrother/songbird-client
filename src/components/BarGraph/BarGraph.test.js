import React from 'react';
import renderer from 'react-test-renderer';

import BarGraph from './BarGraph';

describe('BarGraph', () => {
  let props;

  beforeEach(() => {
    props = {
      data: [
        { label: 'Openness', opposite: 'Shyness', value: 10 },
        { label: 'Extrovert', opposite: 'Introvert', value: -30 },
      ]
    };
  });

  it('renders correctly', () => {
    const tree = renderer.create(<BarGraph {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
