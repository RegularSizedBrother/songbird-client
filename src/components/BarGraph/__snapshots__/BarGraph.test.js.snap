// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarGraph renders correctly 1`] = `
<div
  className="recharts-responsive-container"
  id="undefined"
  style={
    Object {
      "height": 500,
      "maxHeight": undefined,
      "minHeight": undefined,
      "minWidth": undefined,
      "width": "100%",
    }
  }
>
  <div />
</div>
`;
